test/Autofac.Diagnostics.DotGraph.Test/Autofac.Diagnostics.DotGraph.Test.csproj

```diff
@@ -0,0 +1,32 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <AssemblyOriginatorKeyFile>../../Autofac.snk</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>

```

Comment:
Do we need to sign test projects?  If we don't need to, then we can avoid embedding the public key in the AssemblyInfo of the main project.

Reply:
I think you need to sign them to allow `InternalsVisibleTo` to work. However, there's been so much churn on whether you even need to/should sign assemblies at all, with the weird .NET Core default public signing thing and all that... I don't know anymore. I admit I "cargo culted" my way into this one - it works for core Autofac, so I stole it.

Reply:
It does appear if I strong name the main `Autofac.Diagnostics.DotGraph` assembly then in `InternalsVisibleTo` you must have the reference _also_ be strong-named or you get a CS1726 warning. I tried changing it to:

```c#
[assembly: InternalsVisibleTo("Autofac.Diagnostics.DotGraph.Test")]
```

...and I ended up with CS1726: `Friend assembly reference 'Autofac.Diagnostics.DotGraph.Test' is invalid. Strong-name signed assemblies must specify a public key in their InternalsVisibleTo declarations.`

Reply:
Understood.